### DISCUSSION TITLE
Integration with Zint barcode library

### DISCUSSION THREAD

**Discussion by ramkumarkb** [typst/typst#1643] [Ideas]
Usecase: In most of the batch processing systems that generate PDF, there is a need to create barcodes so that the printing / enveloping system can read these barcodes.

Zint: Zint is very good and high qaulity barcode library - with `libzint` being available in the permissible BSD lincense.
https://github.com/zint/zint 

Idea: To create a Rust binding for `libzint` and the use this to create a `barcoding` package within the `Typst` package ecosystem.

References: MRDF - https://patents.google.com/patent/US5777883A/en

**Comment by laurmaedje** [comment]
This seems like a good candidate for a WebAssembly plugin once Typst supports such a thing.

**Comment by ramkumarkb** [comment]
@laurmaedje - Thank you for the reply. 

However, am little confused about the need for `wasm`. If one creates a rust binding on `libzint` and then creates a `typst` package for barcode - will it not work?

**Reply by laurmaedje** [reply]
The only planned way to call native code in Typst is by compiling it to WebAssembly and loading it as a plugin.

**Comment by ramkumarkb** [comment]
@laurmaedje - Ok, noted. I am sure there are some valid reasons for doing this (I am not very familiar with the Typst compiler code).

But the trade-offs for this decision would be unable to integrate with some very good native libs (e.g. Zint, SQLIte, Curl, MetaPost etc). One can always argue why a Typesetting system should integrate with libs like Curl or SQLite - A batch oriented typesetting system always throw up some usecases, where such integration could be very useful.

**Reply by laurmaedje** [reply]
There is a lot of discussion about this topic here: #1056
